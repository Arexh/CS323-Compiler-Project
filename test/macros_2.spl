#define add(x, x); (x) = (x) + 1;
#define 
#define r return 0;
#define $max calmax();
#define PI 3.14
#undef PI
#define PI 3.1415926


int main(){
    add(a, PI);
    r
}