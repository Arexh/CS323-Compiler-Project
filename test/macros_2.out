Error macro at Line 1: duplicate parameter
Error DEFINE at Line 2: empty macro
Error type A at Line 4: unknown lexeme $max
Program (1)
  ExtDefList (10)
    ExtDef (10)
      Specifier (10)
        TYPE: int
      FunDec (10)
        ID: main
        LP
        RP
      CompSt (10)
        LC
        StmtList (11)
          Stmt (11)
            Exp (11)
              ID: add
              LP
              Args (11)
                Exp (11)
                  ID: a
                COMMA
                Args (11)
                  Exp (11)
                    FLOAT: 3.1415926
              RP
            SEMI
          StmtList (12)
            Stmt (12)
              RETURN
              Exp (12)
                INT: 0
              SEMI
        RC
